src/encoding/armor.js

```diff
@@ -247,6 +247,9 @@ function splitChecksum(text) {
   if (lastEquals >= 0) {
     body = text.slice(0, lastEquals);
     checksum = text.slice(lastEquals + 1);
+    if (checksum.substr(0, 6) === '\n-----') {
```

Comment:
Huh, I think it works because your armored string has no trailing newline, so the regular expression doesn't match the trailer. If you had a trailing newline on your armored string it would break.

I think we should solve this by doing a `text = text.trim() + "\n"` transformation at the beginning of the dearmor() function. Then the trailer will always match, and my logic above will work. Sound good?